include/crm/pengine/pe_types.h

```diff
@@ -46,7 +47,7 @@ typedef struct resource_object_functions_s {
     /* parameter result must be free'd */
     char *(*parameter) (pe_resource_t*, pe_node_t*, gboolean, const char*,
                         pe_working_set_t*);
-    void (*print) (pe_resource_t*, const char*, long, void*);
+    void (*print) (pe_resource_t*, const char*, long, void*, pcmk__output_t*);
```

Comment:
This header and print function are part of the library's public API, whereas output.h and pcmk__output_t are private API. This would break public API backward compatibility, and since the private headers aren't installed, this header couldn't be used in an installed environment.

I'm thinking the best way around this will be to leave the existing print functions as they are, \and add the new print functions as an entirely new function. We could maybe call it output instead of print. The output argument could be a void* to avoid needing the private type in the public header. In other words, I'm thinking:

    void (*print) (pe_resource_t*, const char*, long, void*);
    void (*output) (pe_resource_t*, const char*, long, void*, void*);

except the new function should be added to the end of the struct, again for backward API compatibility.

That would lead to some code duplication for the time being, but we would deprecate the print function and remove it next time we break API compatibility (probably a few years from now). The pacemaker code base itself would only use the new function.